proto/api.go

```diff
@@ -51,41 +52,187 @@ const (
 	isWrite                // write cmds go through raft and must be proposed on leader
 	isTxnWrite             // txn write cmds start heartbeat and are marked for intent resolution
 	isRange                // range commands may span multiple keys
+	isReverse              // reverse commands traverse ranges in descending direction
 )
 
 // IsAdmin returns true if the request requires admin permissions.
 func IsAdmin(args Request) bool {
+	if br, ok := args.(*BatchRequest); ok {
```

Comment:
Oh, I did this very differently (and I think considerably simpler) but to the same effect. We need to coordinate here. Damn. Let me separate out that part of the PR and send it out.